app/lib/utils/queryMapper.js

```diff
@@ -16,6 +16,9 @@ function getLocationHeading(query) {
         return 'Where would you like to collect your test kit?';
       }
     }
+    if (query.type.localeCompare(constants.SERVICE_TYPES.online, 'en', { sensitivity: 'base' }) === 0) {
```

Comment:
The verbose 'localeCompare' logic could be extracted into a utility function to improve readability, i.e.
`areEqual(query.type, constants.ServiceTypes.online)`.

As the type in the query string was previously generated by the application itself, could a `===` be used instead?

Although not part of this work, the four 'or' checks starting at line 6 could be changed to a check to see if the query.origin is contained within an array of the four valid options.

Reply:
As the type in the query string was previously generated by the application itself, could a === be used instead? - It is possible for users to fiddle with the URL, hence the localeCompare.

I will add the rest of the suggestions in.

Reply:
If they fiddle with the url won't they get the 'unknown URL' page, which would seem acceptable.

Reply:
agrees, but this handles the situation of type=professional and type=PROFESSIONAL which should be equivalent and not give an unknown URL